diff --git a/nixos/release-combined.nix b/nixos/release-combined.nix
index 416d21ddc5fc..993dd4242e5b 100644
--- a/nixos/release-combined.nix
+++ b/nixos/release-combined.nix
@@ -23,15 +23,7 @@ let
 
   pkgs = import ./.. { };
 
-  removeMaintainers =
-    set:
-    if builtins.isAttrs set then
-      if (set.type or "") == "derivation" then
-        set // { meta = removeAttrs (set.meta or { }) [ "maintainers" ]; }
-      else
-        lib.mapAttrs (n: v: removeMaintainers v) set
-    else
-      set;
+  removeMaintainers = lib.id;
 
 in
 rec {
